gtsam/linear/LinearSolver.h

```diff
@@ -0,0 +1,70 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file    LinearSolver.h
+ * @brief   Common Interface for Linear Solvers
+ * @author  Fan Jiang
+ */
+
+#pragma once
+
+#include <gtsam/linear/LinearSolverParams.h>
+#include <gtsam/linear/VectorValues.h>
+
+namespace gtsam {
+
+/** Common Interface Class for all linear solvers */
+class GTSAM_EXPORT LinearSolver {
+ public:
+  LinearSolver(LinearSolver &) = delete;
+
+  // TODO: Remove this and use trait functions instead?
```

Comment:
Change to TODO(fan)

Reply:
irrelevant, since these functions no longer exist